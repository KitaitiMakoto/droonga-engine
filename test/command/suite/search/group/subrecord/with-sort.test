# -*- js -*-
#@require-catalog-version 2
#@disable-logging
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Threads",
    "key": "ruby",
    "values": {
      "name": "Ruby"
    }
  }
}
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Threads",
    "key": "groonga",
    "values": {
      "name": "Groonga"
    }
  }
}
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Comments",
    "key": "1",
    "values": {
      "thread": "ruby",
      "content": "Hi! I'm a new Rubyist!"
    }
  }
}
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Comments",
    "key": "2",
    "values": {
      "thread": "groonga",
      "content": "Groonga is great!"
    }
  }
}
#@enable-logging
{
  "type": "search",
  "dataset": "Droonga",
  "body": {
    "queries": {
      "comments": {
        "source": "Comments",
        "groupBy": {
          "key": "thread",
          "maxNSubRecords": 10
        },
        "sortBy": ["_key"],
        "output": {
          "elements": ["count", "records"],
          "limit": -1,
          "attributes": [
            "_key",
            "_nsubrecs",
            {
              "label": "items",
              "source": "_subrecs",
              "attributes": [
                "content"
              ]
            }
          ]
        }
      }
    }
  }
}
