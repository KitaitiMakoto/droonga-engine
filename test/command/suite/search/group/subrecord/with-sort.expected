[
  "droonga.message",
  0,
  {
    "inReplyTo": "request-id",
    "statusCode": 200,
    "type": "search.result",
    "body": {
      "comments": {
        "count": 2,
        "records": [
          [
            "groonga",
            1,
            [
              [
                "Groonga is great!"
              ]
            ]
          ],
          [
            "ruby",
            1,
            [
              [
                "Hi! I'm a new Rubyist!"
              ]
            ]
          ]
        ]
      }
    }
  }
]
